students/JoeNunnelley/session01/break_me.py

```diff
@@ -0,0 +1,42 @@
+#! /usr/bin/env python3
+
+""" break me : output error handling
+    Errors:
+    NameError, TypeError, SyntaxError, AttributeError
+"""
+
+def name_error():
+    print("the name error function")
+    try:
+        printx("hello")
+    except NameError:
+        print('name exception caught')
+
+def type_error():
+    print('the type error function')
+    try:
+       mylist = [4,5]
+       print("2" + mylist)
+    except TypeError:
+        print('type error caught')
+
+def attribute_error():
+    print("the attribute error function")
+    try:
+        print("joe".x)
+    except AttributeError:
+        print('attribute error caught')
+
+def syntax_error():
+    print("the syntax error function")
+    try:
+       print("uncomment line below to see an actual SyntaxError")
+       #4.split()
+    except SyntaxError:
+        print('syntax error caught')
+
+
+name_error()
+type_error()
+attribute_error()
+syntax_error()
```

Comment:
We weren't expecting you to catch the errors -- but nice job!

Did you notice that you never caught the syntax error though? that one is raised when teh code is parsed -- so it can't actually run and be caught by a try: except block.